loosen date-time validation

ankhyk opened the pull request:
should not report error when date-time value is just missing a Z in the end

ankhyk committed:
loosen date-time validation

ankhyk committed:
recode, fix test error, update snapshot

raych1 commented:
/azp run

raych1 commented on lib/swaggerValidator/ajvSchemaValidator.ts:
```diff
@@ -229,6 +229,23 @@ const shouldSkipError = (error: ErrorObject, cxt: SchemaValidateContext) => {
     return true;
   }
 
+  // If a response has property which x-ms-secret value is "true" in post we can skip this error
```
Comment needs to be changed.

raych1 commented on lib/swaggerValidator/ajvSchemaValidator.ts:
```diff
@@ -229,6 +229,23 @@ const shouldSkipError = (error: ErrorObject, cxt: SchemaValidateContext) => {
+  if (
+    cxt.isResponse &&
+    keyword === "format" &&
+    schema === "date-time" &&
```
Can you confirm if schema will have the value of 'date-time-rfc1123'?

raych1 commented on test/modelValidation/swaggers/specification/dateTime/dateTime.json:
```diff
@@ -0,0 +1,259 @@
+          "description": "The active directory user name."
+        },
+        "createTime": {
+          "format": "date-time",
```
Can you have another case of 'date-time-rfc1123'?

ankhyk committed:
Some updates
- update comments
- validate date-time-rfc1123
- add test for date-time-rfc1123

ankhyk committed:
update numberOfSpecs

ankhyk committed:
update validation logic

ankhyk committed:
Merge branch 'develop' into loosen_date-time_validation

ankhyk committed:
update numberOfSpecs

raych1 commented on lib/swaggerValidator/ajvSchemaValidator.ts:
```diff
@@ -229,6 +229,26 @@ const shouldSkipError = (error: ErrorObject, cxt: SchemaValidateContext) => {
 
+  // If payload has property with date-time parameter and its value is valid except missing "Z" in the end we can skip this error
+  if (
+    cxt.isResponse &&
```
`cxt.isResponse` can also be removed coz the check applies for request too.

ankhyk committed:
update validation logic

raych1 reviewed (commented):
Can you modify the swagger and payload to cover `datetime` property validation in request either?

ankhyk commented:
> Can you modify the swagger and payload to cover `datetime` property validation in request either?

The payload already has date-time property in both request and response

ankhyk committed:
add change log
